src/plugins/navsat_map/NavSatMap.cc

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "NavSatMap.hh"
+
+#include <algorithm>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include <ignition/common/Console.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+
+#include "ignition/gui/Application.hh"
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class NavSatMapPrivate
+  {
+    /// \brief List of topics publishing navSat messages.
+    public: QStringList topicList;
+
+    /// \brief Holds data to set as the next navSat
+    public: msgs::NavSat navSatMsg;
+
+    /// \brief Node for communication.
+    public: transport::Node node;
+
+    /// \brief Mutex for accessing navSat data
+    public: std::recursive_mutex navSatMutex;
+  };
+}
+}
+}
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+NavSatMap::NavSatMap()
+  : Plugin(), dataPtr(new NavSatMapPrivate)
+{
+}
+
+/////////////////////////////////////////////////
+NavSatMap::~NavSatMap()
+{
+}
+
+/////////////////////////////////////////////////
+void NavSatMap::LoadConfig(const tinyxml2::XMLElement *_pluginElem)
+{
+  // Default name in case user didn't define one
+  if (this->title.empty())
+    this->title = "Navigation satellite map";
+
+  std::string topic;
+  bool topicPicker = true;
+
+  // Read configuration
+  if (_pluginElem)
+  {
+    if (auto topicElem = _pluginElem->FirstChildElement("topic"))
+      topic = topicElem->GetText();
+
+    if (auto pickerElem = _pluginElem->FirstChildElement("topic_picker"))
+      pickerElem->QueryBoolText(&topicPicker);
+  }
+
+  if (topic.empty() && !topicPicker)
+  {
+    ignwarn << "Can't hide topic picker without a default topic." << std::endl;
+    topicPicker = true;
+  }
+
+  this->PluginItem()->setProperty("showPicker", topicPicker);
+
+  if (!topic.empty())
+  {
+    this->SetTopicList({QString::fromStdString(topic)});
+    this->OnTopic(QString::fromStdString(topic));
+  }
+  else
+    this->OnRefresh();
+}
+
+/////////////////////////////////////////////////
+void NavSatMap::ProcessMessage()
+{
+  std::lock_guard<std::recursive_mutex> lock(this->dataPtr->navSatMutex);
+
+  this->newMessage(this->dataPtr->navSatMsg.latitude_deg(),
+      this->dataPtr->navSatMsg.longitude_deg());
```

Comment:
What if `latitude/longitude_deg` is not set in the new message?

Reply:
Ah that's a tough one, protobuf doesn't make a distinction between `double`s that are not set or set to zero. I think it's not unreasonable to assume that these values must be set here. 

We could determine a convention like passing NaN when the fields are unknown. I see that the ROS message specifies NaNs for a missing altitude, but not for missing lat/lon :thinking: https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/msg/NavSatFix.msg

Reply:
> I think it's not unreasonable to assume that these values must be set here

:+1: Sounds good to me